config/config-istio.yaml

```diff
@@ -19,8 +19,16 @@ metadata:
   namespace: knative-serving
 data:
   # TODO(nghia): Extract the .svc.cluster.local suffix into its own config.
-  # Default Knative Gateway after v0.3. It points to the Istio standard istio-ingressgateway,
-  # instead of a custom one that we used pre-0.3.
+
+  # Default Knative Gateway after v0.3. It points to the Istio
+  # standard istio-ingressgateway, instead of a custom one that we
+  # used pre-0.3.
   gateway.knative-ingress-gateway: "istio-ingressgateway.istio-system.svc.cluster.local"
-  # This is the pre-0.3 shared Gateway.  We leave it here for backward compatibility.
+  # This is the pre-0.3 shared Gateway.  We leave it here for backward
+  # compatibility.
   gateway.knative-shared-gateway: "knative-ingressgateway.istio-system.svc.cluster.local"
+  # A cluster local gateway to allow pods outside of the mesh to access
+  # Services and Routes not exposing through an ingress.  If the users
+  # do have a service mesh setup, this isn't required.  If the users
+  # don't have a service mesh, uncomment to use.
+  # local-gateway.cluster-local-gateway: "cluster-local-gateway.istio-system.svc.cluster.local"
```

Comment:
I'm curious as to why we aren't standing this up by default to make it easier to use?

Reply:
Put another way, could we strive for a world in 0.4 where we have a cluster-local gateway as the ONLY place things are exposed until you configure an external domain (in place of the "example.com" hack we have today)?

We can then encourage folks to configure `$(gateway IP).nlp.io` or something to get domain-based routing.

Reply:
Done